src/hyperspectral/hyperspectral/target/rpca.py

```diff
@@ -0,0 +1,123 @@
+import math
+
+import numpy as np
+import scipy
+from tqdm.autonotebook import tqdm
+
+
+def mad(M):
+    return np.median(np.abs(M - np.median(M, axis=0)), axis=0)
+
+
+def rpca_grid(data, max_dim=None, n_c=5, n_g=11, S=mad, sufficient=None):
```

Comment:
It might be useful to include a citation for this (so that one can read in prose what you are accomplishing in code).